org.lflang/src/org/lflang/generator/c/CCompiler.java

```diff
@@ -73,77 +78,212 @@ public class CCompiler {
      * @param targetConfig The current target configuration.
      * @param fileConfig The current file configuration.
      * @param errorReporter Used to report errors.
-     * @param CppMode Indicate if the compilation should happen in C++ mode
+     * @param cppMode Whether the generated code should be compiled as if it
+     * were C++.
      */
     public CCompiler(
-            TargetConfig targetConfig,
-            FileConfig fileConfig,
-            ErrorReporter errorReporter,
-            boolean CppMode
-            ) {
-        this(targetConfig, fileConfig, errorReporter);
-        this.CppMode = CppMode;
-    }
-
-    /**
-     * Create an instance of CCompiler.
-     *
-     * @param targetConfig The current target configuration.
-     * @param fileConfig The current file configuration.
-     * @param errorReporter Used to report errors.
-     */
-    public CCompiler(
-            TargetConfig targetConfig,
-            FileConfig fileConfig,
-            ErrorReporter errorReporter) {
+        TargetConfig targetConfig,
+        FileConfig fileConfig,
+        ErrorReporter errorReporter,
+        boolean cppMode
+    ) {
         this.fileConfig = fileConfig;
         this.targetConfig = targetConfig;
         this.errorReporter = errorReporter;
         this.commandFactory = new GeneratorCommandFactory(errorReporter, fileConfig);
-        this.CppMode = false;
+        this.cppMode = cppMode;
     }
 
     /**
-     * Run the C compiler by directly invoking a C compiler (gcc by default).
-     *
-     * @param file The source file to compile without the .c extension.
-     * @param noBinary If true, the compiler will create a .o output instead of a binary.
-     *  If false, the compile command will produce a binary.
-     * @param generator An instance of GenratorBase, only used to report error line numbers
+     * Run the C compiler by invoking cmake and make.
+     * @param generator An instance of GeneratorBase, only used to report error line numbers
      *  in the Eclipse IDE.
      *
      * @return true if compilation succeeds, false otherwise.
      */
     public boolean runCCompiler(
-        String file,
-        boolean noBinary,
         GeneratorBase generator,
         LFGeneratorContext context
     ) throws IOException {
-        if (noBinary && context.getMode() == LFGeneratorContext.Mode.STANDALONE) {
-            errorReporter.reportError("Did not output executable; no main reactor found.");
-        }
-        LFCommand compile = compileCCommand(file, noBinary);
+        // Set the build directory to be "build"
```

Comment:
Much of the code here is not new; it is just copied from `CCmakeCompiler`, which is now gone.

Reply:
:+1: